### PR TITLE
SSE-2666 Set up a KMS key to decrypt verification codes from Cognito

### PR DESCRIPTION
**Pull request by rafal-surowiec-co**

### CODE REVIEW

File: backend/cognito/src/handlers/decrypter.ts
Comment (CharlesIC) [inline backend/cognito/src/handlers/decrypter.ts:1]: This should live outside of `handlers` as this is not a lambda handler

File: backend/cognito/src/handlers/decrypter.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {toByteArray} from "base64-js";
+import {buildClient, CommitmentPolicy, KmsKeyringNode} from "@aws-crypto/client-node";
+
+const {decrypt} = buildClient(CommitmentPolicy.REQUIRE_ENCRYPT_ALLOW_DECRYPT);
+const generatorKeyId = process.env.KEY_ALIAS;
+const keyIds = process.env.KEY_ARN ? [process.env.KEY_ARN] : [];
```
[the remark this answers is not in the document]
Comment (rafal-surowiec-co) [reply backend/cognito/src/handlers/decrypter.ts:6]: Throw an error

File: backend/cognito/tests/handlers/security-code-text-message-sender.test.ts [hunk 2]
```diff
@@ -8,6 +10,24 @@ const notifyClientMock = jest.fn().mockImplementation(() => {
...
+const buildClientMock = jest.fn().mockImplementation(() => {
+    return {
+        decrypt: decryptMock
+    };
+});
+const CommitmentPolicyMock = jest.fn();
+const KmsKeyringNodeMock = jest.fn();
+jest.mock("@aws-crypto/client-node", () => {
+    return {buildClient: buildClientMock, CommitmentPolicy: CommitmentPolicyMock, KmsKeyringNode: KmsKeyringNodeMock};
+});
```
[the remark this answers is not in the document]
Comment (rafal-surowiec-co) [reply backend/cognito/tests/handlers/security-code-text-message-sender.test.ts:29]: addressed

File: backend/cognito/src/handlers/decrypter.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rafal-surowiec-co) [reply backend/cognito/src/handlers/decrypter.ts:6]: Error will be thrown. Unit test added.

File: backend/cognito/src/handlers/decrypter.ts
Comment (rafal-surowiec-co) [reply backend/cognito/src/handlers/decrypter.ts:1]: addressed

File: backend/cognito/src/handlers/decrypter.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+const keyring = new KmsKeyringNode({generatorKeyId, keyIds});
+
+export default async function decryptCode(code: string): Promise<string> {
```
[the remark this answers is not in the document]
Comment (rafal-surowiec-co) [reply backend/cognito/src/handlers/decrypter.ts:9]: addressed

File: backend/cognito/src/handlers/decrypter.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {toByteArray} from "base64-js";
+import {buildClient, CommitmentPolicy, KmsKeyringNode} from "@aws-crypto/client-node";
+
+const {decrypt} = buildClient(CommitmentPolicy.REQUIRE_ENCRYPT_ALLOW_DECRYPT);
```
[the remark this answers is not in the document]
Comment (rafal-surowiec-co) [reply backend/cognito/src/handlers/decrypter.ts:4]: Best option for us is actually REQUIRE_ENCRYPT_REQUIRE_DECRYPT.

**Review by CharlesIC — APPROVED** [review]

**Pull request merged**